### PR TITLE
Adding example of "revealed" polygon hole with outline

### PR DESCRIPTION
**Pull request by langsmith**
This pr adds an example of an outlined polygon hole in a larger shaded area in San José (cc @1ec5). Inspired by 
Screen Shot 2019-05-07 at 4 02 32 PM

ezgif com-resize (3)

### CODE REVIEW

**Comment by langsmith** [comment]
Example like this could be used to create something like @ClareTrainor 's map 👇 😃 


Screen Shot 2019-04-04 at 10 55 24 AM

File: MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/dds/RevealedPolygonHoleOutlineActivity.java [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+      innerList.add(LineString.fromLngLats(pointList));
+
+      List<List<Point>> megaInnerList = new ArrayList<>();
+      megaInnerList.add(pointList);
+
+      loadedStyle.addSource(new GeoJsonSource(FILL_SOURCE_ID,
+        Polygon.fromOuterInner(LineString.fromLngLats(OUTER_POLYGON_COORDINATES), innerList)));
+
+      loadedStyle.addSource(new GeoJsonSource(LINE_SOURCE_ID,
+        Polygon.fromLngLats(megaInnerList)));
```
Comment (LukasPaczos) [inline MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/dds/RevealedPolygonHoleOutlineActivity.java]: LineLayer should have a LineString as a source.

File: MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/dds/RevealedPolygonHoleOutlineActivity.java [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+  }
+
+  private void addStyling(@NonNull List<Point> pointList) {
+    if (mapboxMap.getStyle() != null) {
+      Style loadedStyle = mapboxMap.getStyle();
+
+      List<LineString> innerList = new ArrayList<>();
+      innerList.add(LineString.fromLngLats(pointList));
+
+      List<List<Point>> megaInnerList = new ArrayList<>();
```
Comment (LukasPaczos) [inline MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/dds/RevealedPolygonHoleOutlineActivity.java]: This list seems obsolete if you can use `LineString` as a geometry for the `LineLayer`.

File: MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/dds/RevealedPolygonHoleOutlineActivity.java [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+      @Override
+      public void onStyleLoaded(@NonNull Style style) {
+        new LoadGeoJson(RevealedPolygonHoleOutlineActivity.this).execute();
+      }
+    });
+  }
+
+  private void addStyling(@NonNull List<Point> pointList) {
+    if (mapboxMap.getStyle() != null) {
+      Style loadedStyle = mapboxMap.getStyle();
```
[the remark this answers is not in the document]
Comment (langsmith) [reply MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/dds/RevealedPolygonHoleOutlineActivity.java]: ✔️ 

https://github.com/mapbox/mapbox-android-demo/blob/84504b92b95298681b6379813e211703e9f07861/MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/dds/RevealedPolygonHoleOutlineActivity.java#L99-L123

File: MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/dds/RevealedPolygonHoleOutlineActivity.java (hunk 1 above)
Comment (langsmith) [reply MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/dds/RevealedPolygonHoleOutlineActivity.java]: ✔️ 

https://github.com/mapbox/mapbox-android-demo/blob/84504b92b95298681b6379813e211703e9f07861/MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/dds/RevealedPolygonHoleOutlineActivity.java#L103-L110

File: MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/dds/RevealedPolygonHoleOutlineActivity.java (hunk 2 above)
Comment (langsmith) [reply MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/dds/RevealedPolygonHoleOutlineActivity.java]: Removed

**Comment by langsmith** [comment]
All feedback addressed, @LukasPaczos . Anything else?

**Comment by langsmith** [comment]
>Do you think it'd make sense to restrict the camera to the outer bounding box?

I also added camera bounds. 

ezgif com-resize (3)

**Comment by jealalbar** [comment]
jl.joker

**Pull request merged**